File: Java/Object Validator/lib/src/main/java/com/quicksilver/objectvalidator/runtime/In.java
package com.quicksilver.objectvalidator.runtime;

import java.util.HashSet;

public class In extends Condition {
    private final String[] args;

    public In(boolean reversed, String fieldExpression, String... args) {
        super(reversed, fieldExpression);
        this.args = args;
    }

    @Override
    protected boolean isFulfilledBy(Object value, String fullFieldExpression, HashSet<String> passedFields, HashSet<String> failedFields) {
        if (value == null) {
            for (String arg : args)
                if (arg == null)
                    return true;
            return false;
        }
        for (String arg : args)
            if (value.toString().equals(arg))
                return true;
        return false;
    }
}

File: C Sharp/Object Validator/Runtime/In.cs
namespace Quicksilver.ObjectValidator.Runtime;
class In(bool reversed, string? fieldExpression, string?[] args) : Condition(reversed, fieldExpression)
{
    private readonly string?[] args = args;

    protected override bool IsFulfilledBy(object? value, string? fullFieldExpression, HashSet<string> passedFields, HashSet<string> failedFields)
    {
        if (value == null) {
            foreach (string? arg in args)
                if (arg == null)
                    return true;
            return false;
        }
        foreach (string? arg in args)
            if (value.ToString() == arg)
                return true;
        return false;
    }
}